### PR TITLE
shameful old branch that should have been merged long ago

### PR DESCRIPTION
**Pull request by matchings**
This branch started out as changes to scripts that I use to run cluster jobs, which shouldn't affect anyone else's work

However there are also a few improvements to other main functions here and there that should be reviewed more carefully (especially anything in `data_access.loading`). 

In addition, I have added several useful functions in` data_access.utilities` to add columns to the `experiment_table` (all automatically applied to the output of `loading.get_platform_paper_experiment_table`), as well as functions to filter experiments and cells according to our various selection criteria we have discussed. These parts of the code should be reviewed carefully by anyone who will use it. 

I have merged in `update_loading_nwb `and master recently, so there shouldn't be any conflicts there

### CODE REVIEW

File: scripts/run_create_multi_session_df.py [hunk 1]
```diff
@@ -1,35 +1,51 @@
...
 
 import visual_behavior.data_access.loading as loading
+from allensdk.brain_observatory.behavior.behavior_project_cache import VisualBehaviorOphysProjectCache as bpc
 
+# python file to execute on cluster
 python_file = r"/home/marinag/visual_behavior_analysis/scripts/create_multi_session_df.py"
 
-jobdir = '/allen/programs/braintv/workgroups/nc-ophys/Marina/ClusterJobs/JobRecords'
+# conda environment to use
+conda_environment = 'visual_behavior_sdk'
```
Comment (yavorska-iryna) [inline scripts/run_create_multi_session_df.py:11]: should this be hardcoded?

**Review by yavorska-iryna — APPROVED** [review]
quick review. I didn't get into all of the details.

File: visual_behavior/data_access/loading.py [hunk 2]
```diff
@@ -107,36 +122,28 @@ def get_ophys_glm_dir():
...
-# LOAD MANIFEST FILES (TABLES CONTAINING METADATA FOR BEHAVIOR & OPHYS DATASETS) FROM SDK CACHE (RECORD OF AVAILABLE DATASETS)
-
-
-def get_cache_dir():
-    """Get directory of data cache for analysis - this should be the standard cache location"""
-    cache_dir = r"//allen/programs/braintv/workgroups/nc-ophys/visual_behavior/2020_cache/production_cache"
-    return cache_dir
-
-
 def get_manifest_path():
```
Comment (alexpiet) [inline visual_behavior/data_access/loading.py:125]: Can you document here that "manifest" points to the production cache

File: visual_behavior/data_access/loading.py [hunk 3]
```diff
@@ -165,63 +171,101 @@ def get_released_ophys_experiment_table(exclude_ai94=True):
...
+    cache_dir = get_platform_analysis_cache_dir()
+    cache = bpc.from_s3_cache(cache_dir=cache_dir)
+    experiment_table = cache.get_ophys_experiment_table()
+
+    # remove 4x2 and Ai94 data
+    experiment_table = experiment_table[(experiment_table.project_code != 'VisualBehaviorMultiscope4areasx2d') &
+                                        (experiment_table.reporter_line != 'Ai94(TITL-GCaMP6s)')].copy()
+
+    # add cell type and binned depth columms for plot labels
+    experiment_table = utilities.add_cell_type_column(experiment_table)
```
Comment (alexpiet) [inline visual_behavior/data_access/loading.py:190]: Can you make loading all these extra columns optional? I use this function for several things that don't need all that extra stuff and its slow.

File: visual_behavior/data_access/loading.py [hunk 4]
```diff
@@ -367,8 +405,59 @@ def get_second_release_candidates():
     return release_candidates
 
 
+def get_extened_stimulus_presentations(stimulus_presentations, licks, rewards, running_speed, eye_tracking):
+    """
+    Takes SDK stimulus presentations table and adds a bunch of useful columns by incorporating data from other tables
```
Comment (alexpiet) [inline visual_behavior/data_access/loading.py:410]: Can you document what columns get added

**Review by alexpiet — APPROVED** [review]

**Pull request merged**